### ISSUE TITLE
Biome Selector

### ISSUE THREAD

**Issue by adamdw499** [punk4eva/DungeonGenerator#10]
Add a basic biome selection mechanic.

Requirements:
1) Materials need to be coherent with the biome system.
2) Filters, materials, room choice, room placement needs to be dependant on the biome.
3) Wide variety of biomes.

**Comment by adamdw499** [comment]
The BiomeProcessor has been implemented and tested slightly. The only noticeable flaws are the current economical consideration of the materials and the fact that the minimum technology requirement of the biome is being used in generating the material curves rather than the technology level of a society. 3 new Filters have also been added and only 2-4 for stone slab and slate are remaining for the current build to be complete. Next commit I will add the remaining filters and fix these two issues.

**Comment by adamdw499** [comment]
Resolved the aforementioned issue regarding the use of the technological level of a society in the material selection calculations.

**Comment by adamdw499** [comment]
Added slate filters and recoloured thatch a little. The next thing is to add stone slab textures. This will probably be done programmatically rather than as an image in order to test how extensible that method is compared to the tedium of drawing graphics.

**Comment by adamdw499** [comment]
Added Slab filter. The dungeon is starting to look better and better. I think it would benefit from many more textures, but that is a job for Build II. I've also moved the technology of the society into its own separate class so that future attributes of the society may be put there as well. As for the handling of the monetary value of the materials, the material's price is meant to be emulated by the combination of factors already in the biome, and the strength of the economy is an emergent property of the sociability and biome. Thus I'm declaring this programming checklist, and Build I finished. Next build I might add:
- More tiles (one for each biome)
- More biomes
- More attributes for the society (ruin level, aggressiveness, magic level, etc.)
- Integrate grass and water colour with biome more.
